packages/core/upload/admin/src/pages/App/MediaLibrary.js

```diff
@@ -169,78 +169,85 @@ export const MediaLibrary = () => {
 
           {!canRead && <NoPermissions />}
 
+          {folderCount === 0 && assetCount === 0 && (
+            <EmptyAssets
+              action={
+                canCreate &&
+                !isFiltering && (
+                  <Button
+                    variant="secondary"
+                    startIcon={<Plus />}
+                    onClick={toggleUploadAssetDialog}
+                  >
+                    {formatMessage({
+                      id: getTrad('header.actions.add-assets'),
+                      defaultMessage: 'Add new assets',
+                    })}
+                  </Button>
+                )
+              }
+              content={
+                // eslint-disable-next-line no-nested-ternary
+                isFiltering
+                  ? formatMessage({
+                      id: getTrad('list.assets-empty.title-withSearch'),
+                      defaultMessage: 'There are no elements with the applied filters',
+                    })
+                  : canCreate
+                  ? formatMessage({
+                      id: getTrad('list.assets.empty'),
+                      defaultMessage: 'Upload your first assets...',
+                    })
+                  : formatMessage({
+                      id: getTrad('list.assets.empty.no-permissions'),
+                      defaultMessage: 'The asset list is empty',
+                    })
+              }
+            />
+          )}
+
           {canRead && (
             <Stack spacing={8}>
-              {folders?.length > 0 && !isFiltering && (
+              {folderCount > 0 && (
                 <FolderList
                   folders={folders}
                   onChangeFolder={handleChangeFolder}
                   onEditFolder={handleEditFolder}
                   onSelectFolder={selectOne}
                   selectedFolders={selected.filter(({ type }) => type === 'folder')}
-                  title={formatMessage({
-                    id: getTrad('list.folders.title'),
-                    defaultMessage: 'Folders',
-                  })}
+                  title={
+                    (((isFiltering && assetCount > 0) || !isFiltering) &&
+                      formatMessage({
+                        id: getTrad('list.folders.title'),
+                        defaultMessage: 'Folders',
+                      })) ||
+                    ''
+                  }
```

Comment:
I'm not sure if we should do this. I think if folders are displayed, we should always display the title too. Let's check in with @maevalienard tomorrow.

Reply:
also synced with Maeva on that, I should have sent you the message, will do that tomorrow!